packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -456,7 +457,17 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                             cachedSnapshot = await this.fetchSnapshot(hostSnapshotOptions);
                         }
                     }
-                    event.end({ method });
+                   const props = { method };
+                   if (props.method === "cache") {
+                       const snapshotCachedEntry = cachedSnapshot as ISnapshotCachedEntry;
```

Comment:
Can we have
```typescript
const props = {};
let cachedSnapshot: ISnapshotContents | ISnapshotCachedEntry | undefined;
const cachedSnapshotP: Promise<ISnapshotCachedEntry | undefined> = this.epochTracker.get(createCacheSnapshotKey(this.odspResolvedUrl)).then((value) => {
    if (value !== undefined) {
       props.cacheEntryAge = ...;
    }
    return value;
};
```
I think that would be enough, but we can always reset props.cacheEntryAge if method != "cache" (this is only needed if both promises were completed essentially at the same time, but Promise.race() choose network over cache).